ir/src/tests/mod.rs

```diff
@@ -0,0 +1,462 @@
+// TODO: add checks for the correctness of the AirIR that is built.
+#[cfg(test)]
+mod tests {
```

Comment:
since we are already inside `tests` module, you don't need another `tests` module here.

Reply:
Ah, sorry, fixed.